packages/shorebird_cli/test/src/commands/patch/patch_android_command_test.dart

```diff
@@ -1085,5 +1085,168 @@ flavors:
       expect(exitCode, equals(ExitCode.success.code));
       verifyNever(() => logger.confirm(any()));
     });
+
+    group('when --sign flag is passed', () {
+      test('succeeds and set the hash signature', () async {
+        when(() => argResults['sign']).thenReturn(true);
+
+        when(
+          () => shorebirdProcess.run(
+            'flutter',
+            any(),
+            runInShell: any(named: 'runInShell'),
+          ),
+        ).thenAnswer((_) async {
+          // Ensure we're building with the correct flutter revision.
+          expect(shorebirdEnv.flutterRevision, equals(release.flutterRevision));
+          return flutterBuildProcessResult;
+        });
+
+        final findSignHashResult = MockShorebirdProcessResult();
+        when(() => findSignHashResult.exitCode)
+            .thenReturn(ExitCode.success.code);
+        when(
+          () => shorebirdProcess.runSync(
+            'ls',
+            ['sign_hash'],
+            runInShell: true,
+          ),
+        ).thenReturn(findSignHashResult);
+
+        final signHashResult = MockProcessResult();
+        when(() => signHashResult.exitCode).thenReturn(ExitCode.success.code);
+        when(() => signHashResult.stdout).thenReturn('Signature');
+        when(
+          () => shorebirdProcess.runSync(
+            './sign_hash',
+            any(),
+            runInShell: true,
+          ),
+        ).thenReturn(signHashResult);
+
+        setUpProjectRoot();
+        setUpProjectRootArtifacts();
+
+        final exitCode = await runWithOverrides(command.run);
+        expect(exitCode, ExitCode.success.code);
+        final captured = verify(
+          () => codePushClientWrapper.publishPatch(
+            appId: appId,
+            releaseId: release.id,
+            platform: releasePlatform,
+            track: track,
+            patchArtifactBundles: captureAny(named: 'patchArtifactBundles'),
+            metadata: const CreatePatchMetadata(
+              releasePlatform: releasePlatform,
+              usedIgnoreAssetChangesFlag: false,
+              hasAssetChanges: false,
+              usedIgnoreNativeChangesFlag: false,
+              hasNativeChanges: false,
+              linkPercentage: null,
+              environment: BuildEnvironmentMetadata(
+                shorebirdVersion: packageVersion,
+                operatingSystem: operatingSystem,
+                operatingSystemVersion: operatingSystemVersion,
+                xcodeVersion: null,
+              ),
+            ),
+          ),
+        ).captured;
+
+        final bundles =
+            (captured.first as Map<Arch, PatchArtifactBundle>).values;
+        for (final bundle in bundles) {
+          expect(bundle.hashSignature, equals('Signature'));
+        }
+      });
+
+      test(
+        "don't break if no sign_hash command is found (do no harm)",
```

Comment:
I don't think this is actually what we want to do. If a user specifies that the patch should be signed, we should not allow the generation of an unsigned patch.

Reply:
Speaking to the "do no harm" value, I think it would be far more harmful to generate the wrong patch without the user knowing than to not generate a patch

Reply:
So, you mean that we should stop the command execution and return a non zero exit code?

Reply:
I think so. If the user specifically asks for a signed patch and we're unable to generate one, I would consider that to be an error.

Reply:
makes sense! Will change